File: Backend/MyApp.CoreService/DTOs/Responses/AccountResponse.cs
using MyApp.CoreService.Enums;

namespace MyApp.CoreService.DTOs.Responses;

public record AccountResponse(
    int Id,
    int OwnerId,
    decimal Balance,
    string Currency,
    AccountStatus Status,
    DateTimeOffset CreatedAt,
    DateTimeOffset? ClosedAt
);

File: Mobile/components/account/src/main/java/nekit/corporation/loan_shared/data/datasource/remote/model/AccountResponse.kt
package nekit.corporation.loan_shared.data.datasource.remote.model

import kotlinx.serialization.Contextual
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class AccountResponse(
    val id: Int,
    val ownerId: Int,
    val balance: Double,
    val currency: String,
    val status: AccountStatus,
    @SerialName("createdAt") @Contextual val createdAt: OffsetDateTime,
    @SerialName("closedAt")  @Contextual val closedAt: OffsetDateTime? = null
)